src/librustc/traits/mod.rs

```diff
@@ -537,6 +537,17 @@ pub fn normalize_param_env_or_error<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
         let region_scope_tree = region::ScopeTree::default();
         let free_regions = FreeRegionMap::new();
+
+        // FIXME. We should really... do something with these region
+        // obligations. But this call just continues the older
+        // behavior (i.e., doesn't cause any new bugs), and it would
+        // take some further refactoring to actually solve them. In
+        // particular, we would have to handle implied bounds
+        // properly, and that code is currently largely confined to
+        // regionck (though I made some efforts to extract it
+        // out). -nmatsakis
+        let _ = infcx.ignore_region_obligations(body_id);
```

Comment:
I think wfcheck already checks that these obligations hold (that's it, that the param env is WF if its normalized predicates hold), and this code should be going away anyway with lazy normalization.

Reply:
Yes, this is why I didn't try to fix this case.

Reply:
ok